### PR TITLE
AWS: Tell S3 bucket name and how to recover if deployment bucket does not exist

### LINKED ISSUE #5699: The specified bucket does not exist: Should tell me the bucket name
# This is a Bug Report

## Description

* What went wrong?

```
  Serverless Error ---------------------------------------
 
  The specified bucket does not exist
```

* What did you expect should have happened?

I expect the error message to tell me what "specified bucket" is so that I can further debug the issue. I never explicitly specified a bucket, so I'm not sure what I should be looking for. I think that would make the error more useful.

* What was the config you used?

```yml
service: my-service

provider:
  name: aws
  runtime: python3.6
  timeout: ${env:LAMBDA_TIMEOUT}

functions:
  erezlife:
    handler: serverless.handler
    events:
      - sqs: ${env:SQS_QUEUE_ARN}

plugins:
  - serverless-python-requirements

custom:
  pythonRequirements:
    noDeploy: []
```

* What stacktrace or error message from your provider did you see?

```
Serverless: Generated requirements from /home/jon/devel/erezlife/requirements.txt in /home/jon/devel/erezlife/.serverless/requirements.txt...
Serverless: Installing requirements from /home/jon/devel/erezlife/.serverless/requirements/requirements.txt ...
Serverless: Packaging service...
Serverless: Excluding development dependencies...
Serverless: Injecting required Python packages to package...
 
  Serverless Error ---------------------------------------
 
  The specified bucket does not exist
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information -----------------------------
     OS:                     linux
     Node Version:           8.12.0
     Serverless Version:     1.35.1
```

Similar or dependent issues:
N/A

## Additional Data

* ***Serverless Framework Version you're using***: 1.35.1
* ***Operating System***: Fedora Linux 27
* ***Stack Trace***: N/A
* ***Provider Error messages***: See above

### PR DESCRIPTION
**Pull request by exoego**
## What did you implement:

Closes #5699

## How did you implement it:

Translate the "The specified bucket does not exist" error into more detailed error containing below:

* Bucket name
* stack name
* Possible recovery operation.

## How can we verify it:

1. `npm i -g exoego/serverless#fix-5699;`
1. `sls deploy` anything
1. Delete the deployment bucket manually.
1. `sls deploy`
1. Verify that new error message is shown.

## Todos:

- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

**Comment by eahefnawy** [comment]
another great contribution @exoego 🙌 ... thanks a lot for your hard work! 💪 

Please let me know if this PR is ready for review, we'd love to see test coverage for this 😊

**Comment by exoego** [comment]
@eahefnawy It's ready 😄

File: lib/plugins/aws/deploy/lib/checkForChanges.js [hunk 1]
```diff
@@ -41,7 +41,17 @@ module.exports = {
...
-    ).then((result) => {
+    ).catch((reason) => {
+      if (!_.includes(reason.message, 'The specified bucket does not exist')) {
+        return BbPromise.reject(reason);
+      }
+      const stackName = this.provider.naming.getStackName();
+      return BbPromise.reject(new this.serverless.classes.Error([
+        `The serverless deployment bucket "${params.Bucket}" does not exist.`,
+        `Create it manually if you want to reuse the CloudFormation stack "${stackName}",`,
+        'or delete the stack if it is no longer required.',
```
Comment (eahefnawy) [inline lib/plugins/aws/deploy/lib/checkForChanges.js]: @exoego what you mean "delete the stack if it is no longer required"? 🤔

Comment (exoego) [reply lib/plugins/aws/deploy/lib/checkForChanges.js]: If the deploy bucket is removed accidentally, `sls remove` can not remove the stack.
So, if user want to delete the stack which is no longer required, the stack should be manually deleted.

**Comment by krlozadan** [comment]
Hello, I know this has been closed but I have the same issue when the bucket was deleted and I'm trying to upload the service again. This is the error that I got:
```
Serverless: DOTENV: Loading environment variables from .env:
Serverless:      - JWT_SECRET
Serverless: Packaging service...
Serverless: Excluding development dependencies...
Serverless: Uploading CloudFormation file to S3...
 
  Serverless Error ---------------------------------------
 
  The specified bucket does not exist
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information ---------------------------
     OS:                     darwin
     Node Version:           10.15.3
     Serverless Version:     1.46.1
```

**Pull request merged**